### ISSUE TITLE
slash behavior on Windows

### ISSUE THREAD

**Issue by jeremybernstein** [zlib-ng/minizip-ng#636]
I may be doing something wrong, or overlooking a configuration option, but it appears that, on Windows systems, `MZ_PATH_SLASH_PLATFORM` is always going to be set to '\\'. This presents problems for archives created on Windows, but extracted on macOS or Unix, where '/' is expected (not to be an intolerable pedant, but the [.zip format description](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) specifies '/' in section 4.4.17:

```
       4.4.17.1 The name of the file, with optional relative path.
       The path stored MUST NOT contain a drive or
       device letter, or a leading slash.  All slashes
       MUST be forward slashes '/' as opposed to
       backwards slashes '\' for compatibility with Amiga
       and UNIX file systems etc.  If input came from standard
       input, there is no file name field.  
```

Mostly this makes `mz_zip_writer_add_path()` unusable on Windows (since it uses `mz_path_combine()` which then uses the platform slash), so it's something which can be worked around, but having an option to override this behavior would be welcome.

Thanks for considering it.

**Comment by nmoinvaz** [comment]
Maybe a change here:
https://github.com/zlib-ng/minizip-ng/blob/47b8449fec2c7a575a506ecbe6399ad0603b5992/mz_zip.c#L790

**Comment by nmoinvaz** [comment]
Merged to `develop` in #644.
